stag-library/src/test/java/com/vimeo/stag/KnownTypeAdaptersTest.java

```diff
@@ -450,8 +452,36 @@ public void testForPrimitiveBooleanTypeAdapter() throws Exception {
 
         // call the TypeAdapter#read method
         boolean readValue = KnownTypeAdapters.PrimitiveBooleanTypeAdapter.read(new JsonReader(new StringReader(jsonString)), false);
-
         Assert.assertEquals(value, readValue);
+
+        String booleanString = "true";
+        // call the TypeAdapter#read method
+        readValue = KnownTypeAdapters.PrimitiveBooleanTypeAdapter.read(new JsonReader(new StringReader(booleanString)), false);
+        Assert.assertEquals(true, readValue);
+
+        booleanString = "false";
+        // call the TypeAdapter#read method
+        readValue = KnownTypeAdapters.PrimitiveBooleanTypeAdapter.read(new JsonReader(new StringReader(booleanString)), false);
+        Assert.assertEquals(false, readValue);
+
+        booleanString = "abcde"; // not valid, throw exception
+        // call the TypeAdapter#read method
+        readValue = KnownTypeAdapters.PrimitiveBooleanTypeAdapter.read(new JsonReader(new StringReader(booleanString)), false);
+        Assert.assertEquals(false, readValue);
```

Comment:
This test throws an exception and the rest of the cases don't run as method execution is halted. You should either break this case and the `booleanString = "2";` case out into their own test methods that are annotated with `@Test (expected = MalformedJsonException.class)`, or you need to write the failing tests as follows:

```java
boolean exceptionWasThrown = false;
try {
booleanString = "abcde"; // not valid, throw exception
// call the TypeAdapter#read method
readValue = KnownTypeAdapters.PrimitiveBooleanTypeAdapter.read(new JsonReader(new StringReader(booleanString)), false);
} catch (MalformedJsonException exception) {
    exceptionWasThrown = true;
}
Assert.assertTrue(exceptionWasThrown);
```